crates/ruff/src/rules/flake8_bandit/rules/hardcoded_sql_expression.rs

```diff
@@ -0,0 +1,105 @@
+use once_cell::sync::Lazy;
+use regex::Regex;
+use ruff_macros::{define_violation, derive_message_formats};
+use rustpython_parser::ast::{Expr, ExprKind, Operator};
+
+use super::super::helpers::string_literal;
+use crate::ast::helpers::{any_over_expr, unparse_expr};
+use crate::ast::types::Range;
+use crate::checkers::ast::Checker;
+use crate::registry::Diagnostic;
+use crate::violation::Violation;
+
+static SQL_REGEX: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(r"(?i)(select\s.*from\s|delete\s+from\s|insert\s+into\s.*values\s|update\s.*set\s)")
+        .unwrap()
+});
+
+define_violation!(
+    /// ### What it does
+    /// Checks for strings that resemble SQL statements involved in some form
+    /// string building operation.
+    ///
+    /// ### Why is this bad?
+    /// SQL injection is a common attack vector for web applications. Unless care
+    /// is taken to sanitize and control the input data when building such
+    /// SQL statement strings, an injection attack becomes possible.
```

Comment:
This section implies that sanitizing inputs is alright ... when actually directly interpolating user input into SQL queries should always be avoided (no matter if sanitized or not).

You should instead use "parameter binding" ... also called "server-side binding", where the parameters are sent separately from the query to the server, see e.g. [psycopg2 server-side binding](https://www.psycopg.org/psycopg3/docs/basic/from_pg2.html#server-side-binding).

So I think it would make sense to update this section accordingly.